suite/tests/api/drdecode_aarch64.c

```diff
@@ -57,6 +57,41 @@ test_disasm(void)
         pc = disassemble_with_info(GD, pc, STDOUT, false /*no pc*/, true);
 }
 
+static void
+test_mov_instr_addr(void)
+{
+#if !defined(DR_HOST_NOT_TARGET)
+    instrlist_t *ilist = instrlist_create(GD);
+    instr_t *callee = INSTR_CREATE_label(GD);
+    instrlist_append(
+        ilist,
+        XINST_CREATE_move(GD, opnd_create_reg(DR_REG_X2), opnd_create_reg(DR_REG_X30)));
+    instrlist_insert_mov_instr_addr(GD, callee, (byte *)ilist, opnd_create_reg(DR_REG_X0),
+                                    ilist, NULL, NULL, NULL);
+    instrlist_append(ilist, INSTR_CREATE_blr(GD, opnd_create_reg(DR_REG_X0)));
+    instrlist_append(ilist, INSTR_CREATE_ret(GD, opnd_create_reg(DR_REG_X2)));
+    instrlist_append(ilist, callee);
+    instrlist_insert_mov_immed_ptrsz(GD, 0xdeadbeef, opnd_create_reg(DR_REG_X20), ilist,
+                                     NULL, NULL, NULL);
+    instrlist_append(ilist, XINST_CREATE_return(GD));
+
+    uint gencode_max_size = 1024;
+    byte *generated_code =
+        (byte *)allocate_mem(gencode_max_size, ALLOW_EXEC | ALLOW_READ | ALLOW_WRITE);
+    assert(generated_code != NULL);
+    instrlist_encode(GD, ilist, generated_code, true);
+    protect_mem(generated_code, gencode_max_size, ALLOW_EXEC | ALLOW_READ);
+
+    ((void (*)(void))generated_code)();
+    unsigned long x20;
+    asm("mov %0,x20" : "=r"(x20));
```

Comment:
x20 is callee-saved and this seems a little fragile in that it could mess up this compiled code.  How about writing the sentinel to x0, the return value, instead?  That seems safer.

Reply:
Done.